### DISCUSSION TITLE
Display Queue with Retry For Failed Downloads

### DISCUSSION THREAD

**Discussion by Vesperindustrial** [BlafKing/sd-civitai-browser-plus#181] [Ideas]
This is somewhat related to the bug issue I filed (https://github.com/BlafKing/sd-civitai-browser-plus/issues/180) But I was thinking that being able to actually see the queue in a list (with an option for clearing the list once done, or manually like in browsers) would be helpful to identify if any downloads fail. Currently if something fails like in the bug issue I filed, then the only way I would know it did (along with the rest of the queue that never starts) is if I manually go back to that page, or I dont find it in my folder afterwards. As mentioned in the title this would probably also include a button for retry or or at least to open the model page to hand download it if required.

**Comment by BlafKing** [comment]
Thanks for the suggestion!
That sounds good to me.

It'll probably take a while to implement, but this is something that I will definitely put on the to-do list! :)

**Comment by BlafKing** [comment]
There is now a queue tab that displays all the models in the queue, you can rearrange the models and remove any of them.

I didn't have time to implement a retry button when a model fails, since I'm going on holiday now and wanted to publish something since I won't be able to work on the extension for a while.

The queue system likely still contains some bugs since I didn't have time to thoroughly test it

**Reply by Vesperindustrial** [reply]
Thanks so much, I'll give it a try soon! Enjoy your holidays!
